server/chat-plugins/customavatars.js

```diff
@@ -0,0 +1,102 @@
+// Original plugin from https://github.com/CreaturePhil/Showdown-Boilerplate/blob/master/chat-plugins/customavatar.js.
+// Credits to CreaturePhil and the other listed contributors.
+// updated for the main server by Maxalexanderpi and Hoeenhero.
+*/
+'use strict';
+
+/** @type {typeof import('../../lib/fs').FS} */
+const path = require('path');
+const fs = require('fs');
+const https = require('https');
+
+const AVATAR_PATH = 'config/avatars/';
+
+const VALID_EXTENSIONS = ['.png'];
+
+function downloadImage(image_url, name, extension) {
+	let req = https.get(image_url, res => {
+		 res.setEncoding('binary');
```

Comment:
No need to ``setEncoding()``. It actually hurts performance by converting the *binary* data to a ``latin-1``-encoded string.